src/bridges/uniswap/UniswapBridge.sol

```diff
@@ -0,0 +1,368 @@
+            _tokenOut
+        );
+
+        if (percentage1 + percentage2 != 100) revert InvalidPercentageAmounts();
+
+        path.percentage2 = percentage2;
+        path.splitPath2 = splitPath2;
+        path.minPrice = _decodeMinPrice(_encodedPath >> SPLIT_PATHS_BIT_LENGTH);
+    }
+
+    /**
+     * @notice A function which returns a percentage of input going through the split path and the split path encoded
+     *         in a format compatible with Uniswap router.
+     * @param _tokenIn - Input ERC20 token
+     * @param _encodedSplitPath - Encoded split path (in the last 19 bits of uint64)
+     * @param _tokenOut - Output ERC20 token
+     * @return percentage - A percentage of input going through the corresponding split path
+     * @return splitPath - A split path encoded in a format compatible with Uniswap router
+     */
+    function _decodeSplitPath(
+        address _tokenIn,
+        uint64 _encodedSplitPath,
+        address _tokenOut
+    ) internal view returns (uint256 percentage, bytes memory splitPath) {
+        uint64 fee3 = _encodedSplitPath & FEE_MASK;
+        uint64 middleToken2 = (_encodedSplitPath >> 2) & TOKEN_MASK;
+        uint64 fee2 = (_encodedSplitPath >> 5) & FEE_MASK;
+        uint64 middleToken1 = (_encodedSplitPath >> 7) & TOKEN_MASK;
+        uint64 fee1 = (_encodedSplitPath >> 10) & FEE_MASK;
+        percentage = _encodedSplitPath >> 12;
+
+        if (middleToken1 != 0 && middleToken2 != 0) {
+            splitPath = abi.encodePacked(
+                _tokenIn,
+                _getFeeTier(fee1),
+                _getMiddleToken(middleToken1),
+                _getFeeTier(fee2),
+                _getMiddleToken(middleToken2),
+                _getFeeTier(fee3),
+                _tokenOut
+            );
+        } else if (middleToken1 != 0) {
+            splitPath = abi.encodePacked(
+                _tokenIn,
+                _getFeeTier(fee1),
+                _getMiddleToken(middleToken1),
+                _getFeeTier(fee3),
+                _tokenOut
+            );
+        } else if (middleToken2 != 0) {
+            splitPath = abi.encodePacked(
+                _tokenIn,
+                _getFeeTier(fee2),
+                _getMiddleToken(middleToken2),
+                _getFeeTier(fee3),
+                _tokenOut
+            );
+        } else {
+            splitPath = abi.encodePacked(_tokenIn, _getFeeTier(fee3), _tokenOut);
+        }
+    }
+
+    /**
+     * @notice A function which converts minimum price in a floating point format to integer.
+     * @param _encodedMinPrice - Encoded minimum price (in the last 26 bits of uint64)
+     * @return minPrice - Minimum acceptable price represented as an integer
+     */
+    function _decodeMinPrice(uint64 _encodedMinPrice) internal pure returns (uint256 minPrice) {
+        // 21 bits significand, 5 bits exponent
+        uint64 significand = _encodedMinPrice >> 5;
```

Comment:
Why are you doing this with `uint64`? The `minPrice` will overflow if larger than `uint64` because the values are not being cast. Use `uint256` instead.

Reply:
You are right. Will fix. Thanks 👍

Reply:
It seems that solidity compiler is smart enough that it knows that the result is uint256 and for this reason it doesn't overflow. Pretty surprised tbh.